### PR TITLE
MAINT: Fix issues involving elif conditions

### PR DESCRIPTION
**Pull request by donno2048**
Fix two elif conditions and remove unnecessary import

### CODE REVIEW

File: scipy/io/arff/arffread.py [hunk 1]
```diff
@@ -273,7 +273,7 @@ def _get_date_format(atrv):
             if "yyyy" in pattern:
                 pattern = pattern.replace("yyyy", "%Y")
                 datetime_unit = "Y"
-            elif "yy":
+            elif "yy" in pattern:
```
Comment (tylerjereddy) [inline scipy/io/arff/arffread.py:276]: For this change here, I should mention that there is both an open issue and an attempt to provide the same fix with a regression test:

https://github.com/scipy/scipy/issues/11864
https://github.com/scipy/scipy/pull/11888

Probably my fault for making it overly complicated and asking for a regression guard (which was proving much harder than the fairly-simple fix)

**Review by tylerjereddy — COMMENTED** [review]
The change in `scipy/stats/stats.py` seems straightforward to me. The `io` change was attempted before with a regression guard so I should at least mention that here.

File: scipy/io/arff/arffread.py (hunk 1 above)
Comment (donno2048) [reply scipy/io/arff/arffread.py]: I reverted this commit... Thanks!

**Review by rgommers — APPROVED** [review]
Straightforward fix now, merging. Thanks @donno2048

**Pull request merged**